### PR TITLE
Color, Tint refactoring

### PR DESCRIPTION
**Pull request by starwed**
It seemed a bit silly to have two components which are basically the same.  (I think one issue was that IE8 didn't support `rgba()` syntax for background color?)

This patch eliminates Tint completely, adding its functionality to Color.  The existing approach of using CSS works for both DOM and Canvas Color entities, but
-  wouldn't work for WebGL -- unlike canvas, it has no idea about color names.  
-  didn't play well with Tint  

So I also added some functionality to Crafty for turning CSS-style string references to colors into the raw RGB values. 

This is the function `Crafty.assignColor`.  If you give it a hex string ("#FF0000") a color name ("red") or something odder (like an "hlsa()" declaration) it'll try to resolve it to rgb values.
- If it's a hex code, it just parses the hex directly.  Works for the optional short form
- If it's an `rgb` or `rgba` declaration, that'll also be parsed directly to rgb.
- If it's an unknown type, it'll assign the string to an invisible div, and then parse the computed color.  For efficiency, this will be stored in a dictionary object.  So things like "lightsalmon" will be computed correctly, but with a small one-time performance hit.
- The dictionary is prepopulated by a short list of the 17 CSS2 colors like "red", "black", etc.

We could simply store _all_ the standard CSS3 colors in the dictionary.  You can see that commented out in the code.  :)   [This SO thread](http://stackoverflow.com/questions/1573053/javascript-function-to-convert-color-names-to-hex-codes) was super-useful in understanding the different approaches here.

This PR needs refactoring once the changes to tests go in.  And I'm uncertain whether we should include all those extra colors by default.

Also, I can't test this in IE!  I could imagine the whole invisible element thing failing, so would be good to know one way or the other.  Running the included unit tests should be fine.

### CODE REVIEW

**Comment by starwed** [comment]
Ok, rebased against the current develop and updated the tests.

I also removed the commented out code, so this is in a state where it could be merged now.  (Would still be nice if someone could test on IE!)

If we wanted to include the standard CSS3 color names, the code would be something like:

```
var dictionary = {"aliceblue":"#f0f8ff","antiquewhite":"#faebd7","aqua":"#00ffff","aquamarine":"#7fffd4","azure":"#f0ffff",
        "beige":"#f5f5dc","bisque":"#ffe4c4","black":"#000000","blanchedalmond":"#ffebcd","blue":"#0000ff","blueviolet":"#8a2be2","brown":"#a52a2a","burlywood":"#deb887",
        "cadetblue":"#5f9ea0","chartreuse":"#7fff00","chocolate":"#d2691e","coral":"#ff7f50","cornflowerblue":"#6495ed","cornsilk":"#fff8dc","crimson":"#dc143c","cyan":"#00ffff",
        "darkblue":"#00008b","darkcyan":"#008b8b","darkgoldenrod":"#b8860b","darkgray":"#a9a9a9","darkgreen":"#006400","darkkhaki":"#bdb76b","darkmagenta":"#8b008b","darkolivegreen":"#556b2f",
        "darkorange":"#ff8c00","darkorchid":"#9932cc","darkred":"#8b0000","darksalmon":"#e9967a","darkseagreen":"#8fbc8f","darkslateblue":"#483d8b","darkslategray":"#2f4f4f","darkturquoise":"#00ced1",
        "darkviolet":"#9400d3","deeppink":"#ff1493","deepskyblue":"#00bfff","dimgray":"#696969","dodgerblue":"#1e90ff",
        "firebrick":"#b22222","floralwhite":"#fffaf0","forestgreen":"#228b22","fuchsia":"#ff00ff",
        "gainsboro":"#dcdcdc","ghostwhite":"#f8f8ff","gold":"#ffd700","goldenrod":"#daa520","gray":"#808080","green":"#008000","greenyellow":"#adff2f",
        "honeydew":"#f0fff0","hotpink":"#ff69b4",
        "indianred ":"#cd5c5c","indigo":"#4b0082","ivory":"#fffff0","khaki":"#f0e68c",
        "lavender":"#e6e6fa","lavenderblush":"#fff0f5","lawngreen":"#7cfc00","lemonchiffon":"#fffacd","lightblue":"#add8e6","lightcoral":"#f08080","lightcyan":"#e0ffff","lightgoldenrodyellow":"#fafad2",
        "lightgrey":"#d3d3d3","lightgreen":"#90ee90","lightpink":"#ffb6c1","lightsalmon":"#ffa07a","lightseagreen":"#20b2aa","lightskyblue":"#87cefa","lightslategray":"#778899","lightsteelblue":"#b0c4de",
        "lightyellow":"#ffffe0","lime":"#00ff00","limegreen":"#32cd32","linen":"#faf0e6",
        "magenta":"#ff00ff","maroon":"#800000","mediumaquamarine":"#66cdaa","mediumblue":"#0000cd","mediumorchid":"#ba55d3","mediumpurple":"#9370d8","mediumseagreen":"#3cb371","mediumslateblue":"#7b68ee",
        "mediumspringgreen":"#00fa9a","mediumturquoise":"#48d1cc","mediumvioletred":"#c71585","midnightblue":"#191970","mintcream":"#f5fffa","mistyrose":"#ffe4e1","moccasin":"#ffe4b5",
        "navajowhite":"#ffdead","navy":"#000080",
        "oldlace":"#fdf5e6","olive":"#808000","olivedrab":"#6b8e23","orange":"#ffa500","orangered":"#ff4500","orchid":"#da70d6",
        "palegoldenrod":"#eee8aa","palegreen":"#98fb98","paleturquoise":"#afeeee","palevioletred":"#d87093","papayawhip":"#ffefd5","peachpuff":"#ffdab9","peru":"#cd853f","pink":"#ffc0cb","plum":"#dda0dd","powderblue":"#b0e0e6","purple":"#800080",
        "red":"#ff0000","rosybrown":"#bc8f8f","royalblue":"#4169e1",
        "saddlebrown":"#8b4513","salmon":"#fa8072","sandybrown":"#f4a460","seagreen":"#2e8b57","seashell":"#fff5ee","sienna":"#a0522d","silver":"#c0c0c0","skyblue":"#87ceeb","slateblue":"#6a5acd","slategray":"#708090","snow":"#fffafa","springgreen":"#00ff7f","steelblue":"#4682b4",
        "tan":"#d2b48c","teal":"#008080","thistle":"#d8bfd8","tomato":"#ff6347","turquoise":"#40e0d0",
        "violet":"#ee82ee",
        "wheat":"#f5deb3","white":"#ffffff","whitesmoke":"#f5f5f5",
        "yellow":"#ffff00","yellowgreen":"#9acd32"};
```

File: src/color.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        }
+        this.trigger("Invalidate");
+    },
+
+    // draw function for "Color"
+    _drawColor: function(e){
+        if (!this._color) { return; }
+        if (e.type === "DOM") {
+            e.style.backgroundColor = this._color;
+            e.style.lineHeight = 0;
```
Comment (mucaho) [inline src/color.js:168]: what's the lineHeight for? (it was already present before, but still curious :) )

File: tests/color.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	var c = {};
+	Crafty.assignColor("rgba(255, 0, 0, 0.5)", c);
+	equal(c._red, 255, "red is 255");
+	equal(c._green, 0, "green is 0");
+	equal(c._blue, 0, "blue is 0");
+	equal(c._strength, 0.5, "strength is 0.5");
+});
+
+module("Color");
+
```
Comment (mucaho) [inline tests/color.js:64]: maybe also add "color by hex", "color by hex + alpha", "color by rgb", "color by rgba" for completeness

**Comment by mucaho** [comment]
> If we wanted to include the standard CSS3 color names, the code would be something like

Why don't we add all the standard color names by default and forget about the invisible div completely? If user wants ultra advanced color, he can specify hex. aka "keep it simple"

File: src/color.js (hunk 1 above)
Comment (starwed) [inline src/color.js:168]: Yeah, that's weird -- I have no idea!  [Lineheight](https://developer.mozilla.org/en-US/docs/Web/CSS/line-height) should only change things when there's text, I thought.  Probably the best way to find out what it's for is to remove that code.  :)

File: tests/color.js (hunk 2 above)
Comment (starwed) [inline tests/color.js:64]: Well, that's really testing assignColor; here I'm just testing the different method signatures.

I could test the other combinations too, if you still feel it's a good idea.

**Comment by starwed** [comment]
> Why don't we add all the standard color names by default and forget about the invisible div completely? If user wants ultra advanced color, he can specify hex. aka "keep it simple"

As long as the invisible div doesn't harm anything, it would be a nice feature.  It's a guarantee that the color you get matches any valid CSS string.  The less Crafty-specific knowledge the API requires, the friendlier it is.

**Comment by mucaho** [comment]
In theory it sounds great, but I feel like it's a dirty low-level hack (and kudos to the discovery of that trick btw). It's just something I would personally avoid, but I can understand if you want to leave it like that (especially since we already have to sacrifice high-level coding in favor of low-level coding to get performant javascript code).

**Comment by starwed** [comment]
I'm going to go ahead and merge this.  It would be nice if someone could test in IE -- but once it's merged it'll show up in the qunit results, so it'll at least get tested before we do a realease, I guess.

**Pull request merged**